src/main/java/pl/tomaszdziurko/jvm_bloggers/utils/DateTimeUtilities.java

```diff
@@ -22,4 +22,9 @@ public static int daysBetweenDateAndLastFriday(LocalDateTime date) {
         }
     }
 
+    public static LocalDateTime lastPublicationDate(NowProvider nowProvider) {
+        LocalDateTime currentDate = nowProvider.now();
+        int daysSinceLastFriday = daysBetweenDateAndLastFriday(currentDate);
+        return currentDate.minusDays(daysSinceLastFriday).withHour(12).withMinute(0);
+    }
```

Comment:
Maybe we should refactor `NowProvider` to `DateProvider` with methods like `now()` and `lastPublicationDate()`. What's your view?